### PR TITLE
Fix duplicate chords exported by Dolet

### PR DESCRIPTION
**Pull request by miiizen**
This PR provides a fix for the following situation, caused by Dolet not being able to export fretboard diagrams without attached chord symbols.  When fretboard diagrams *without* chord symbols attached are used in Sibelius and then supplemented with staff text we parse into a chord symbol, the following happens: 
image

If two chord symbols with matching chords exist at a tick, this PR favours the symbol with a fret diagram attached, discarding the other.

The PR also adds inference for chord symbols making use of the existing chord symbol parser in `ParsedChord::parse`.  These symbols won't have fret diagrams attached.

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
See #22370, unit tests not running...

**Review by oktophonie — APPROVED** [review]

**Review by mike-spa — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
Do you plan on porting #8530 too? That seems to collide a bit...

File: src/importexport/musicxml/internal/musicxml/musicxmlsupport.h [hunk 1]
```diff
@@ -74,6 +75,23 @@ struct MusicXmlArpeggioDesc {
 };
 typedef std::multimap<int, MusicXmlArpeggioDesc> ArpeggioMap;
 
+struct HarmonyDesc
```
[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/musicxmlsupport.h:78]: HarmonyDescr might cover it better?

[the remark this answers is not in the document]
Comment (mike-spa) [reply src/importexport/musicxml/internal/musicxml/musicxmlsupport.h:78]: Ah ok, then yes it's ok to keep it consistent with the rest of the module :)

**Comment by Jojo-Schmitz** [comment]
@miiizen mind to reply to https://github.com/musescore/MuseScore/pull/22329#issuecomment-2061620043 ?

**Pull request merged**